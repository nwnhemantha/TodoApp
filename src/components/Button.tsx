import React, { Fragment } from 'react'
import { Button } from 'antd';

type Types = "primary" | "text" | "link" | "ghost" | "default" | "dashed";
type Size = "large" | "middle" | "small";

interface Props {
    text: string,
    type?: Types,
    onClick: () => void,
    classNames?: string,
    size?: Size
    danger?: boolean,
}
const Buttons: React.FC<Props> = ({ text, type, onClick, classNames, size, danger }: Props) => {
    return (
        <Fragment>
            <Button 
            danger={danger} 
            size={size} 
            type={type} 
            className={classNames} 
            onClick={onClick}>{text}
        </Button>
        </Fragment>
    )
}

export default Buttons
