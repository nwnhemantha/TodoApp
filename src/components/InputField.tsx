import React, { Fragment } from 'react';
import { Input } from 'antd';

interface Props{
    classNames?: string,
    value: string,
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
}

const InputField: React.FC<Props> = ({ classNames, onChange, value }: Props) =>  {
    return (
        <Fragment>
            <Input placeholder="add your task here ..." className={classNames} onChange={onChange} value={value}/>
        </Fragment>
    )
}

export default InputField
