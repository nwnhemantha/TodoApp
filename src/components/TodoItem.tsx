import React, { Fragment } from 'react';
import { List } from 'antd';
import Button from './Button';

// eslint-disable-next-line no-unused-vars
enum Status { pending, done}

interface Props{
    item: {
        text: string,
        status: number,
    }
    index: number,
    classNames?: string,
    onDelete: (key: number) => void
    onComplete: (key: number, status: Status) => void
}

const TodoItem: React.FC<Props> = ({ item, onDelete, onComplete, index, classNames}: Props) =>  {

    return (
        <Fragment>
            <List.Item className={classNames}>
                <List.Item.Meta
                description={item.text}
                className={item.status === Status.pending? "" : "line-through"}
                />
                <div><Button text={item.status === Status.pending? "Done" : "Not Done"} size="small" onClick={ () => onComplete(index, item.status) } /></div>
                <div><Button text="X" size="small" danger classNames="ml-2" onClick={() => onDelete(index)} /></div>
            </List.Item>
        </Fragment>
    )
}

export default TodoItem
