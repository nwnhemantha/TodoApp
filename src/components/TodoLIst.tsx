import React, { Fragment } from 'react';
import TodoItem from './TodoItem'

// eslint-disable-next-line no-unused-vars
enum Status { pending, done}

interface List {
    text:string,
    status: Status,
}

interface Props {
    list: List[] | undefined,
    onDelete: (key: number) => void,
    onComplete: (key: number, status: Status) => void,
}

const TodoLIst: React.FC<Props> = ({ list, onDelete, onComplete }: Props) => {

    return (
        <Fragment>
            {
                list?.map( (item, key) => <TodoItem onComplete={onComplete} onDelete={onDelete} key={key} index={key} item={item} />)
            }
        </Fragment>
    )
}

export default TodoLIst
