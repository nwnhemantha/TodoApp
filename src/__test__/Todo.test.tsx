import React from 'react';
import Button from '../components/Button';
import InputField from '../components/InputField';
import TodoItem from '../components/TodoItem';
import TodoLIst from '../components/TodoLIst';
import renderer from 'react-test-renderer';

test('Render the Button component correctly', () => {
  const tree = renderer
    .create(<Button text="Add" onClick={jest.fn} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render the InputField component correctly', () => {
  const tree = renderer
    .create(<InputField value="" onChange={jest.fn}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render the TodoItem component correctly', () => {
  const tree = renderer
    .create(<TodoItem item={{text: '', status: 0}} index={0} onComplete={jest.fn} onDelete={jest.fn} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render the TodoLIst component correctly', () => {
  const tree = renderer
    .create(<TodoLIst list={[{text: '', status: 0}]} onComplete={jest.fn} onDelete={jest.fn}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
