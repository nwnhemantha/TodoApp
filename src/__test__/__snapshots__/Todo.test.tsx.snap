// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the Button component correctly 1`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
>
  <span>
    Add
  </span>
</button>
`;

exports[`Render the InputField component correctly 1`] = `
<input
  className="ant-input"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  placeholder="add your task here ..."
  type="text"
  value=""
/>
`;

exports[`Render the TodoItem component correctly 1`] = `
<li
  className="ant-list-item"
>
  <div
    className="ant-list-item-meta"
  >
    
  </div>
  <div>
    <button
      className="ant-btn ant-btn-sm"
      onClick={[Function]}
      type="button"
    >
      <span>
        Done
      </span>
    </button>
  </div>
  <div>
    <button
      className="ant-btn ml-2 ant-btn-sm ant-btn-dangerous"
      onClick={[Function]}
      type="button"
    >
      <span>
        X
      </span>
    </button>
  </div>
</li>
`;

exports[`Render the TodoLIst component correctly 1`] = `
<li
  className="ant-list-item"
>
  <div
    className="ant-list-item-meta"
  >
    
  </div>
  <div>
    <button
      className="ant-btn ant-btn-sm"
      onClick={[Function]}
      type="button"
    >
      <span>
        Done
      </span>
    </button>
  </div>
  <div>
    <button
      className="ant-btn ml-2 ant-btn-sm ant-btn-dangerous"
      onClick={[Function]}
      type="button"
    >
      <span>
        X
      </span>
    </button>
  </div>
</li>
`;
